contracts/InstadappAdapter.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+
+import {IDSA} from "./interfaces/IDSA.sol";
+
+/// @title InstadappAdapter
+/// @author Connext
+/// @notice This contract is inherited by InstadappTarget, it includes the logic to verify signatures
+/// and execute the calls.
+/// @dev This contract is not meant to be used directly, it is meant to be inherited by other contracts.
+/// @custom:experimental This is an experimental contract.
+contract InstadappAdapter is EIP712 {
+  /// Structs
+  /// @dev This struct is used to encode the data for InstadappTarget.cast function.
+  /// @param _targetNames The names of the targets that will be called.
+  /// @param _datas The data that will be sent to the targets.
+  /// @param _origin The address that will be used as the origin of the call.
+  struct CastData {
+    string[] _targetNames;
+    bytes[] _datas;
+    address _origin;
+  }
+
+  /// @dev This struct is used to encode the data that is signed by the auth address.
+  /// The signature is then verified by the verify function.
+  struct Sig {
+    CastData castData;
+    bytes32 salt;
+    uint256 deadline;
+  }
+
+  /// Storage
+  /// @dev This mapping is used to prevent replay attacks.
+  mapping(bytes32 => bool) private sigReplayProtection;
+
+  /// Constants
+  /// @dev This is the typehash for the CastData struct.
+  bytes32 public constant CASTDATA_TYPEHASH =
+    keccak256("CastData(string[] _targetNames,bytes[] _datas,address _origin)");
```

Comment:
For EIP-712, I think we can use just normal variable naming `varName`

Reply:
done